app/scripts/modules/cloudfoundry/src/pipeline/stages/createServiceBindings/CloudFoundryCreateServiceBindingsConfig.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { cloneDeep } from 'lodash';
+import { FormikErrors } from 'formik';
+
+import { IStage, FormValidator, FormikStageConfig, IStageConfigProps } from '@spinnaker/core';
+
+import { CloudFoundryCreateServiceBindingsStageConfigForm } from './CloudFoundryCreateServiceBindingsStageConfigForm';
+
+interface ServiceBindingRequests {
+  serviceInstanceName: String;
+}
+
+export function CloudFoundryCreateServiceBindingsConfig({
+  application,
+  pipeline,
+  stage,
+  updateStage,
+}: IStageConfigProps) {
+  const stageWithDefaults = React.useMemo(() => {
+    return {
+      serviceBindingRequests: [],
+      restageRequired: true,
+      restartRequired: false,
+      credentials: '',
+      region: '',
+      ...cloneDeep(stage),
+    };
+  }, []);
+
+  return (
+    <FormikStageConfig
+      application={application}
+      onChange={updateStage}
+      pipeline={pipeline}
+      stage={stageWithDefaults}
+      validate={validateCloudFoundryCreateServiceBindingsStage}
+      render={(props) => <CloudFoundryCreateServiceBindingsStageConfigForm {...props} />}
+    />
+  );
+}
+
+export function validateCloudFoundryCreateServiceBindingsStage(stage: IStage): FormikErrors<IStage> {
+  const formValidator = new FormValidator(stage);
+
+  formValidator.field('credentials', 'Account').required();
+  formValidator.field('region', 'Region').required();
+  formValidator.field('cluster', 'Cluster').required();
+  formValidator.field('target', 'Target').required();
+
+  formValidator
+    .field('serviceBindingRequests', 'Service Binding Requests')
+    .required()
+    .withValidators((serviceBindingRequests: ServiceBindingRequests[]) => {
+      if (validateServiceBindingRequests(serviceBindingRequests)) {
+        return undefined;
+      }
+      return 'There should be at least one service binding request. At a minimum, each request must have a service instance name.';
+    });
+
+  return formValidator.validateForm();
+}
+
+export function validateServiceBindingRequests(serviceBindingRequests: ServiceBindingRequests[]): boolean {
+  if (serviceBindingRequests?.length < 1) {
+    return false;
+  }
+  for (let i = 0; i < serviceBindingRequests.length; i++) {
```

Comment:
I feel this could be simplified with ["some"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) or ["every"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) like:

```
return serviceBindingRequests.every((req) => req.serviceInstanceName);
```

Reply:
I tried this and it didn't capture when the fields were empty ("") then false. If you want go ahead and give it a shot. I'm always open to simplifying the readability

Reply:
my mistake, your solution also worked. Swapped & Merged :)

Reply:
it should work, just ran these in chrome console
```js
> [].every(x=>x)
true

> [""].every(x => x)
false

> [
	true,
	1,
	{},
].every(x => x)
true
```